### PR TITLE
Bug fix:KubernetesProxy requestHandler interrupted by decorateClusterDetailsWithAuth error

### LINKED ISSUE #17604: 🐛 Bug Report: Kubernetes proxy endpoint doesn't work with google auth
### 📜 Description

Using the Kubernetes proxy endpoint with `google` authProvider results in a 500 and error message.

With config:

```
kubernetes:
  serviceLocatorMethod:
    type: 'multiTenant'
  clusterLocatorMethods:
    - type: 'gke'
      projectId: 'some-project'
      skipTLSVerify: true # Fix this
      skipMetricsLookup: true
```

I'm seeing the following error when requesting pod logs on the main branch

### 👍 Expected behavior

It should authenticate correctly and return the logs

### 👎 Actual Behavior with Screenshots

Screenshot 2023-05-01 at 9 10 28 PM

### 👟 Reproduction steps

I suspect this might be for any client side auth provider for example:
1. starting the backend with:
```
kubernetes:
  serviceLocatorMethod:
    type: 'multiTenant'
  clusterLocatorMethods:
    - type: 'gke'
      projectId: 'some-project'
      skipTLSVerify: true # Fix this
      skipMetricsLookup: true
```

2. apply the dice roller manifests

3. navigate to http://localhost:3000/catalog/default/component/dice-roller/kubernetes

4. Click a cluster > deployment > pod > get logs

5. see the error

### 📃 Provide the context for the Bug.

I suspect it is at [this line](https://github.com/backstage/backstage/blob/master/plugins/kubernetes-backend/src/kubernetes-auth-translator/GoogleKubernetesAuthTranslator.ts#L42) as the empty object is being pass in when it should be an auth object with the client side token

### 🖥️ Your Environment

_No response_

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by RubenV-dev**
## Hey, I just made a Pull Request!
Fixes #17604, `KubernetesProxy` bug. 

in the case that the request to the proxy endpoint has a `Backstage-Kubernetes-Authorization` header and the cluster in question has a `client-side authProvider`, the `decorateClusterDetailsWithAuth` method was throwing an error which interrupted the execution of the request handler. Express caught this error and served a 500 response. 

Solution was tested following bug reproduction steps included in the above issue.

The following pod-logs are now available
Screenshot 2023-05-04 at 11 16 19 AM

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by jamieklassen** [comment]
Can we get a unit test for this? You could even modify

https://github.com/backstage/backstage/blob/c9eff7b1fac77c6b5cb8ecb5c2b7ea9562adb6c5/plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts#L276

to give the coverage you need, by changing the setup on the mock translator here

https://github.com/backstage/backstage/blob/c9eff7b1fac77c6b5cb8ecb5c2b7ea9562adb6c5/plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts#L310-L315

to reject with an error instead of resolving

**Review by jamieklassen — CHANGES_REQUESTED** [review]
Can we get a unit test for this? You could even modify

https://github.com/backstage/backstage/blob/c9eff7b1fac77c6b5cb8ecb5c2b7ea9562adb6c5/plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts#L276

to give the coverage you need, by changing the setup on the mock translator here

https://github.com/backstage/backstage/blob/c9eff7b1fac77c6b5cb8ecb5c2b7ea9562adb6c5/plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts#L310-L315

to reject with an error instead of resolving.

Also need a changeset of course

**Comment by RubenV-dev** [comment]
> Can we get a unit test for this? You could even modify
> 
> https://github.com/backstage/backstage/blob/c9eff7b1fac77c6b5cb8ecb5c2b7ea9562adb6c5/plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts#L276
> 
> to give the coverage you need, by changing the setup on the mock translator here
> 
> https://github.com/backstage/backstage/blob/c9eff7b1fac77c6b5cb8ecb5c2b7ea9562adb6c5/plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts#L310-L315
> 
> to reject with an error instead of resolving.
> 
> Also need a changeset of course

Woops, changeset added. Yep i will add a unit test to get some coverage.

**Comment by jamieklassen** [comment]
FWIW I don't think I'd describe the cause of the bug as having to do with "overriding" -- in the case that the request to the proxy endpoint has a `Backstage-Kubernetes-Authorization` header, it was not being overridden; rather, in such cases the `decorateClusterDetailsWithAuth` method was throwing an error which interrupted the execution of the request handler. Express caught this error and served a 500 response

**Comment by RubenV-dev** [comment]
Ah, sure thing. I understood it to be more like the `kubernetesProxy` was attempting to overwrite the clusterDetails.serviceAccountToken field using the `decorateCusterDetailsWithAuth` method which throws the error. But no problem i will change the language up to be less confusing. Thanks for the feedback.

**Comment by mclarke47** [comment]
lgtm once we add the test, I'll let @jamieklassen approve when he's ready

**Comment by jamieklassen** [comment]
> Ah, sure thing. I understood it to be more like the `kubernetesProxy` was attempting to overwrite the clusterDetails.serviceAccountToken field using the `decorateCusterDetailsWithAuth` method which throws the error. But no problem i will change the language up to be less confusing. Thanks for the feedback.

I like the change to the PR title, but what feels more important to me is the commit message, which people will see while they're deep in their IDE or doing `git blame`s, etc

**Comment by RubenV-dev** [comment]
> > Ah, sure thing. I understood it to be more like the `kubernetesProxy` was attempting to overwrite the clusterDetails.serviceAccountToken field using the `decorateCusterDetailsWithAuth` method which throws the error. But no problem i will change the language up to be less confusing. Thanks for the feedback.
> 
> I like the change to the PR title, but what feels more important to me is the commit message, which people will see while they're deep in their IDE or doing `git blame`s, etc

commit message changed

File: .changeset/spotty-walls-serve.md
Comment (jamieklassen) [inline .changeset/spotty-walls-serve.md:5]: maybe I'm missing something, but why the `inline code block` syntax here?
```suggestion
Fixed a bug in the Kubernetes proxy endpoint where requests to clusters configured with client-side auth providers would always fail with a 500 status.
```

File: plugins/kubernetes-backend/src/service/KubernetesProxy.ts
Comment (jamieklassen) [inline plugins/kubernetes-backend/src/service/KubernetesProxy.ts:116]: I think you can use a [nullish coalescing operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing) here to tighten it even more?
```suggestion
      req.headers.authorization = req.header(HEADER_KUBERNETES_AUTH)
        ?? `Bearer ${
```

File: .changeset/spotty-walls-serve.md
Comment (RubenV-dev) [reply .changeset/spotty-walls-serve.md:5]: I tend to get a lot of vale errors in changeset messages when mentioning words like kubernetes, so i inline the words to state these are words that make sense to the code-base but might not necessarily make sense in a sentence. Just something ive done as a chore when submitting PR's.

Comment (jamieklassen) [reply .changeset/spotty-walls-serve.md:5]: the word "kubernetes" should be allowed:

https://github.com/backstage/backstage/blob/08bd36717d79dee994e5d7fe1690a889c41b10ea/.github/vale/Vocab/Backstage/accept.txt#L184

The thing about putting words in code blocks is that then they don't get spell-checked at all! This is OK for actual code symbols like class names but "client-side auth provider" is supposed to be a phrase for humans

File: plugins/kubernetes-backend/src/service/KubernetesProxy.ts
Comment (RubenV-dev) [reply plugins/kubernetes-backend/src/service/KubernetesProxy.ts]: Done

File: plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts
Comment (jamieklassen) [inline plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts:446]: I feel this should really be a [400 status](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) -- if the request is missing the cluster header, then it's a "bad request".

File: plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts
Comment (jamieklassen) [inline plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts:481]: To get the right error mapping, I think we could add something like this
```suggestion
    const router = Router();
    router.use('/mountpath', proxy.createRequestHandler({ permissionApi }));
    router.use(errorHandler());
    const app = express().use(router);
```
where `errorHandler` is the function from `@backstage/backend-common` documented [here](https://backstage.io/docs/reference/backend-common.errorhandler/)

**Review by jamieklassen — CHANGES_REQUESTED** [review]

File: plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts
Comment (RubenV-dev) [reply plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts]: Sure thing, Done

File: plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts
Comment (RubenV-dev) [reply plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts]: Done

**Review by jamieklassen — APPROVED** [review]
Thanks for handling my nits!

**Pull request merged**
